src/NexusMods.DataModel/Interprocess/Jobs/InterprocessJob.cs

```diff
@@ -58,6 +61,24 @@ public class InterprocessJob : IInterprocessJob
     {
     }
 
+    /// <summary>
+    /// Create a new job, where the payload is a IMessage
+    /// </summary>
+    /// <param name="jobType"></param>
+    /// <param name="manager"></param>
+    /// <param name="payload"></param>
+    /// <param name="description"></param>
+    /// <typeparam name="T"></typeparam>
+    /// <returns></returns>
+    public static InterprocessJob Create<T>(JobType jobType, IInterprocessJobManager manager, T payload, string description)
+        where T : IMessage
+    {
+        Span<byte> data = stackalloc byte[T.MaxSize];
```

Comment:
The amount of memory allocated with`stackalloc` should be limited (`512` or `1024` are common limits):

```csharp
Span<byte> data = T.MaxSize > 1024 ? new byte[T.MaxSize] : stackalloc byte[T.MaxSize]
```